akka-actor-typed/src/main/scala/akka/actor/typed/BehaviorInterceptor.scala

```diff
@@ -4,40 +4,48 @@
 
 package akka.actor.typed
 
+import scala.reflect.ClassTag
+
 import akka.annotation.{ DoNotInherit, InternalApi }
+import akka.util.BoxedType
 
 /**
  * A behavior interceptor allows for intercepting message and signal reception and perform arbitrary logic -
  * transform, filter, send to a side channel etc. It is the core API for decoration of behaviors. Many built-in
  * intercepting behaviors are provided through factories in the respective `Behaviors`.
  *
- * If the interceptor does keep mutable state care must be taken to create the instance in a `setup` block
- * so that a new instance is created per spawned actor rather than shared among actor instance.
+ * If the interceptor does keep mutable state care must be taken to create a new instance from
+ * the factory function of `Behaviors.intercept` so that a new instance is created per spawned
+ * actor rather than shared among actor instance.
+ *
+ * @param interceptMessageClass Ensures that the interceptor will only receive `O` message types.
+ *                              If the message is not of this class or a subclass thereof it will
+ *                              bypass the interceptor and be continue to the inner behavior untouched.
+ *
+ * @tparam Outer The outer message type – the type of messages the intercepting behavior will accept
+ * @tparam Inner The inner message type - the type of message the wrapped behavior accepts
  *
- * @tparam O The outer message type – the type of messages the intercepting behavior will accept
- * @tparam I The inner message type - the type of message the wrapped behavior accepts
+ * @see [[BehaviorSignalInterceptor]]
```

Comment:
Should we maybe add a note that it is a power user feature and not the first thing you should reach for (given recent discussion)?

Reply:
Ok, we can add something more to above "Many built-in intercepting behaviors are provided through factories in the respective `Behaviors`."

Reply:
added in 362ac824d038d83f7a96be466b02581d2d59b4f3